File: client-sdk/csharp/sdk/src/VertexCache/Sdk/Protocol/Parsers/ValueResponseParser.cs
using VertexCache.Sdk.Results;

namespace VertexCache.Sdk.Protocol.Parsers
{
    public class ValueResponseParser : IResponseParser
    {
        public bool CanParse(string response) =>
            response.StartsWith("+") && response != "+OK" && response != "+PONG" && response != "+(nil)" && response != "+Deleted";

        public VCacheResult Parse(string response) => VCacheResult.SuccessWithRaw(response);
    }
}

File: client-sdk/go/sdk/protocol/parsers/value_response_parser.go
package parsers

import "vertexcache/sdk/results"
import "strings"

type ValueResponseParser struct{}

func (p *ValueResponseParser) CanParse(response string) bool {
	return strings.HasPrefix(response, "+") &&
		response != "+OK" &&
		response != "+PONG" &&
		response != "+(nil)" &&
		response != "+Deleted"
}

func (p *ValueResponseParser) Parse(response string) *results.VCacheResult {
	return results.NewSuccessWithRaw(response)
}
